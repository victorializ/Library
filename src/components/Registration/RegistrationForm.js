import React, { useState } from 'react';

import {
  Button,
  Form,
  Checkbox
} from 'semantic-ui-react';

import { constants } from "../../i18n";

function RegistrationForm({register}) {

  const [ user, setUser ] = useState(
    {
      firstName: '', 
      lastName: '', 
      email: '', 
      password: '', 
      reenter: ''
    }
  );

  const [ checked, setChecked ] = useState(false);

  const handleChange = ({target: {name, value}}) => {
    setUser({...user, [name]: value});
  }

  const isSubmitDisabled = () => {
    return !checked || 
      user.password !== user.reenter || 
      Object.keys(user).some(key => !user[key]);
  }

  return (
    <Form>
      <Form.Field>
        <label>
          {constants.firstName}
        </label>
        <input 
          className='registration__firstName'
          name='firstName'
          value={user.firstName}
          placeholder={constants.firstName} 
          onChange={handleChange}/>
      </Form.Field>
      <Form.Field>
        <label>
          {constants.lastName}
        </label>
        <input 
          className='registration__lastName'
          name='lastName'
          value={user.lastName}
          placeholder={constants.lastName} 
          onChange={handleChange}/>
      </Form.Field>
      <Form.Field>
        <label>
          {constants.email}
        </label>
        <input
          className='registration__email'
          name='email'
          value={user.email}
          placeholder={constants.email} 
          onChange={handleChange}/>
      </Form.Field>
      <Form.Field>
        <label>
          {constants.password}
        </label>
        <input
          className='registration__password'
          type='password'
          name='password'
          value={user.password}
          placeholder={constants.password} 
          onChange={handleChange}/>
      </Form.Field>
      <Form.Field>
        <label>
          {constants.reenter}
        </label>
        <input 
          className='registration__reenter'
          type='password'
          name='reenter'
          value={user.reenter}
          placeholder={constants.reenter} 
          onChange={handleChange}/>
      </Form.Field>
      <Checkbox 
        className='registration__agree'
        checked={checked}
        label={constants.agree}
        onChange={() => setChecked(!checked)}
      />
      <Button 
        className='registration__button'
        onClick={() => register(user)}
        disabled={isSubmitDisabled()}>
        {constants.register}
      </Button>
    </Form>
  );
}

export { RegistrationForm };