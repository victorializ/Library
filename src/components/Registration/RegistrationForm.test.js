import React from 'react';
import { shallow } from 'enzyme';

import { RegistrationForm } from './RegistrationForm';

describe('RegistrationForm component', () => {
    it('renders correctly', () => {
        expect(shallow(<RegistrationForm />)).toMatchSnapshot();
    }); 
});