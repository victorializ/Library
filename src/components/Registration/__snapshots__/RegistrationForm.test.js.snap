// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationForm component renders correctly 1`] = `
<Form
  as="form"
>
  <FormField>
    <label>
      First name
    </label>
    <input
      className="registration__firstName"
      name="firstName"
      onChange={[Function]}
      placeholder="First name"
      value=""
    />
  </FormField>
  <FormField>
    <label>
      Last name
    </label>
    <input
      className="registration__lastName"
      name="lastName"
      onChange={[Function]}
      placeholder="Last name"
      value=""
    />
  </FormField>
  <FormField>
    <label>
      Email
    </label>
    <input
      className="registration__email"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      value=""
    />
  </FormField>
  <FormField>
    <label>
      Password
    </label>
    <input
      className="registration__password"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
  </FormField>
  <FormField>
    <label>
      Reenter password
    </label>
    <input
      className="registration__reenter"
      name="reenter"
      onChange={[Function]}
      placeholder="Reenter password"
      type="password"
      value=""
    />
  </FormField>
  <Checkbox
    checked={false}
    className="registration__agree"
    label="I agree to the Terms and Conditions"
    onChange={[Function]}
    type="checkbox"
  />
  <Button
    as="button"
    className="registration__button"
    disabled={true}
    onClick={[Function]}
  >
    Registration
  </Button>
</Form>
`;
